test/tools/unified-spec-runner/match.ts

```diff
@@ -184,6 +184,9 @@ export function resultCheck(
     } else if (Long.isLong(expected) && typeof actual === 'number') {
       // Long requires special equality check
       expect(expected.equals(actual)).to.be.true;
+    } else if (Number.isNaN(actual) && Number.isNaN(expected)) {
```

Comment:
which test added a case where this is applicable?

Reply:
In addition to the new constructed tests, our existing tests are in EJSON and could give us NaN values like so:
```ts
{"$numberDouble": "Infinity" | "-Infinity" | "NaN"> }
```

While we're here can we add a proper equality check for `0` and `-0`. 
- if expected and actual are both typeof number, and both `=== 0`
- then assert `Object.is(actual, expected)` is true.

```js
Object.is(0, -0) // false
```

Reply:
Added the case for +/- zero!